lib/jekyll/excerpt.rb

```diff
@@ -116,11 +116,29 @@ def place_in_layout?
     def extract_excerpt(doc_content)
       head, _, tail = doc_content.to_s.partition(doc.excerpt_separator)
 
+      # append appropriate closing tag (to a Liquid block), to the "head" if the
+      # partitioning resulted in leaving the closing tag somewhere in the "tail"
+      # partition.
+      if head.include?("{%")
+        head =~ %r!{%\s*(\w+).+\s*%}!
+        tag_name = Regexp.last_match(1)
+
+        if liquid_block?(tag_name) && head.match(%r!{%\s*end#{tag_name}.+\s*%}!).nil?
```

Comment:
What's with the `.+`?

If is necessary, you can get rid of the `\s*`, since `.+` will also swallow any whitespace.

Reply:
nice catch.. `.+` is not needed since end-tags do not take argument(s)..